src/emit.rs

```diff
@@ -296,13 +279,16 @@ impl<'c, H: HugrView> EmitHugr<'c, H> {
     ///
     /// If any LLVM IR declaration which is to be emitted already exists in the
     /// [Module] and it differs from what would be emitted, then we fail.
-    pub fn emit_func(mut self, node: FatNode<'c, FuncDefn, H>) -> Result<Self> {
-        let mut worklist: EmissionSet<'c, H> = [node].into_iter().collect();
-        let pop =
-            |wl: &mut EmissionSet<'c, H>| wl.iter().next().cloned().map(|x| wl.take(&x).unwrap());
+    pub fn emit_func(mut self, node: FatNode<'_, FuncDefn, H>) -> Result<Self> {
+        let mut worklist: EmissionSet = [node.node()].into_iter().collect();
+        let pop = |wl: &mut EmissionSet| wl.iter().next().cloned().map(|x| wl.take(&x).unwrap());
 
         while let Some(x) = pop(&mut worklist) {
-            let (new_self, new_tasks) = self.emit_func_impl(x)?;
+            use crate::fat::FatExt as _;
+            let Some(func) = node.hugr().try_fat(x) else {
```

Comment:
Should a try_fat failure here be a panic? IIUC this wasn't a failure mode before, and now there's a possibility of functions being silently removed from the TODO list without anything having been done